### PR TITLE
[feat] Playlist page/features

### PR DESCRIPTION
**Pull request by Hamdrive**
## Overview

Successfully completed creation of playlist page and implementation of all its features

Changes:
1. User can add video to playlist from explore page and single video page
2. User can create multiple playlists from modal
3. User can create empty playlist from the playlist page itself
4. User can delete entire playlist 
5. User can delete single video inside playlist

# Note to reviewers:
1.  Kindly ignore the watchlater menu option, will be integrated in a later issue
2.  Kindly ignore all styles related changes (if not familiar with MUI) and instead review the logic implemented.

## Previews
image
image
image

### CODE REVIEW

File: src/components/playlistModal/PlaylistModal.jsx [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+  } = useVideo();
+  const [newPlaylist, setnewPlaylist] = useState(false);
+  const [playlistTitle, setplaylistTitle] = useState("");
+
+  const location = useLocation();
+
+  const handleNewPlaylist = (e) => {
+    e.preventDefault(e);
+    setPlaylists(token, playlistTitle, videoDispatch);
+    setplaylistTitle("");
```
Comment (prachi-sahani) [inline src/components/playlistModal/PlaylistModal.jsx]: use camelCase

File: src/components/ExploreCard/ExploreCard.jsx [hunk 2]
```diff
@@ -6,22 +6,54 @@ import {
...
+      name: "Add to Watch Later",
+      icon: <WatchLaterIcon />,
+    },
+    {
+      id: 2,
+      name: "Add to Playlist",
+      icon: <PlaylistAddIcon />,
+      action: () => {
+        handleModalOpen(video);
+        setanchorEl(null);
```
Comment (prachi-sahani) [inline src/components/ExploreCard/ExploreCard.jsx]: use camelCase

File: src/components/ExploreCard/ExploreCard.jsx [hunk 3]
```diff
@@ -6,22 +6,54 @@ import {
...
-  thumbnail,
-  creator,
-  creatorLogo,
-  slug,
-  location,
-  handleIconAction,
+  video,
+  handleModalOpen,
+  // handlePlaylist,
+  // playlist
```
Comment (prachi-sahani) [inline src/components/ExploreCard/ExploreCard.jsx]: remove commented out code, if not required

**Review by prachi-sahani — COMMENTED** [review]
Code is clean
Good use of barrel imports
Liked the implementation of  `EmptyResult` component
Apart from suggestions mentioned below, everything looks fine

Suggestions
 - Add auth check while creating/adding playlist
 - On click of `login as guest`, it should not just fill up the login credentials. Instead the user should also be logged in or the button text should be `Fill Test Credentials`, if you are just populating the email and password.

File: src/pages/singlePlaylist/SinglePlaylist.jsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    deleteVideoFromPlaylist,
+  } = useVideo();
+
+  const {
+    authState: { token },
+  } = useAuth();
+
+  const { slug } = useParams();
+
+  const currentPlaylist = playlists?.find((playlist) => playlist._id === slug);
```
Comment (FarhanMobashir) [inline src/pages/singlePlaylist/SinglePlaylist.jsx]: There is a route for single playlist too. You should use that.

**Review by FarhanMobashir — APPROVED** [review]
Overall the code is good. Added one suggestion. Do also check the suggestion from other reviewer

File: src/pages/singlePlaylist/SinglePlaylist.jsx (hunk 4 above)
Comment (Hamdrive) [reply src/pages/singlePlaylist/SinglePlaylist.jsx]: Yes that is right, will make the necessary changes

File: src/components/ExploreCard/ExploreCard.jsx (hunk 3 above)
Comment (Hamdrive) [reply src/components/ExploreCard/ExploreCard.jsx]: Done

File: src/components/ExploreCard/ExploreCard.jsx (hunk 2 above)
Comment (Hamdrive) [reply src/components/ExploreCard/ExploreCard.jsx]: Done

File: src/components/playlistModal/PlaylistModal.jsx (hunk 1 above)
Comment (Hamdrive) [reply src/components/playlistModal/PlaylistModal.jsx]: Done

**Pull request merged**